projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/routing_policy/statement/Statements.java

```diff
@@ -73,6 +79,41 @@ public Result execute(Environment environment) {
           result.setFallThrough(true);
           break;
 
+        case RemovePrivateAs:
+          {
+            BgpRoute.Builder bgpRouteBuilder = (BgpRoute.Builder) environment.getOutputRoute();
+            List<SortedSet<Integer>> newAsPath =
+                bgpRouteBuilder
+                    .getAsPath()
+                    .stream()
+                    .map(
+                        asSet ->
+                            asSet
+                                .stream()
+                                .filter(AsPath::isNotPrivateAs)
+                                .collect(ImmutableSortedSet.toImmutableSortedSet(Integer::compare)))
+                    .filter(asSet -> !asSet.isEmpty())
+                    .collect(ImmutableList.toImmutableList());
+            bgpRouteBuilder.setAsPath(newAsPath);
+            if (environment.getWriteToIntermediateBgpAttributes()) {
+              BgpRoute.Builder ir = environment.getIntermediateBgpAttributes();
+              List<SortedSet<Integer>> iAsPath =
+                  ir.getAsPath()
+                      .stream()
+                      .map(
+                          asSet ->
+                              asSet
+                                  .stream()
+                                  .filter(AsPath::isNotPrivateAs)
+                                  .collect(
+                                      ImmutableSortedSet.toImmutableSortedSet(Integer::compare)))
+                      .filter(asSet -> !asSet.isEmpty())
+                      .collect(ImmutableList.toImmutableList());
```

Comment:
I would extract everything from L101-111 in a static helper, so we can call the same function twice.